src/scripts/profiling/run_profiling.py

```diff
@@ -43,10 +89,12 @@ def run_profiling(
         output_pyis_file = output_dir / "output.pyisession"
         output_html_file = output_dir / "output.html"
         output_json_file = output_dir / "output.json"
+        output_stat_file = output_dir / "output.stats.json"
```

Comment:
Can't suggest change here as not all lines touched in this PR, but could avoid the repetition / need to keep the extensions in sync here by doing something like
```Python
output_stem = "output" if output_name is None else output_name.stem
output_pyis_file = output_dir / f"{output_stem }.pyisession"
output_html_file = output_dir / f"{output_stem }.html"
output_json_file = output_dir / f"{output_stem }.json"
output_stat_file = output_dir / f"{output_stem}.stats.json"
```